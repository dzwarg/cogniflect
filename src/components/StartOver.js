import React from 'react';
import {Button, Col, Row} from 'react-bootstrap';
import {LinkContainer} from 'react-router-bootstrap';

const StartOver = ({children, questions, startOverHandler}) => (
  <Row>
    <Col xs={12}>
        <p>
          <LinkContainer to="/">
            <Button onClick={startOverHandler(questions)} bsStyle="warning">Start Over</Button>
          </LinkContainer>
          {children}
        </p>
    </Col>
  </Row>
);

export default StartOver;