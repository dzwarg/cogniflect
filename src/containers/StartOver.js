import { connect } from 'react-redux'
import StartOver from '../components/StartOver'
import actions from '../actions'

const mapStateToProps = (state) => ({
  questions: state.appState.get('questions')
})

const mapDispatchToProps = (dispatch) => ({
  startOverHandler: (questions) => (e) => {
    dispatch(actions.set(['assessmentType'], 'individual'))
    dispatch(actions.set(['teamCode'], ''))
    dispatch(actions.set(['userTeamCode'], ''))
    dispatch(actions.set(['questionCursor'], 0))
    dispatch(actions.set(['questions'], questions.map((question) => {
      return question.set('myAnswer', null).set('ourAnswer', null)
    })))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(StartOver)
export {mapStateToProps, mapDispatchToProps}