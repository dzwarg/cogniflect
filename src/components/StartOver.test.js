import React from 'react';
import StartOver from './StartOver';
import {fromJS} from 'immutable';

const initialState = {
  appState: fromJS({
    children: (<p>Hello, world!</p>),
    proceed: null,
    questions: [{
      myAnswer: true,
      ourAnswer: true
    },{
      myAnswer: false,
      ourAnswer: false
    }],
    
  })
};

describe('dumb component', () => {
  it('without crashing', () => {
    const component = StartOver({...initialState, startOverHandler:jest.fn()})
    expect(component).toMatchSnapshot();
  });
});
