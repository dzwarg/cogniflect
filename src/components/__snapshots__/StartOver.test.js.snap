// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dumb component without crashing 1`] = `
<ForwardRef(Bootstrap(Row))>
  <ForwardRef(Bootstrap(Col))
    xs={12}
  >
    <p>
      <LinkContainer
        activeClassName="active"
        exact={false}
        replace={false}
        strict={false}
        to="/"
      >
        <ForwardRef(Bootstrap(Button))
          bsStyle="warning"
        >
          Start Over
        </ForwardRef(Bootstrap(Button))>
      </LinkContainer>
    </p>
  </ForwardRef(Bootstrap(Col))>
</ForwardRef(Bootstrap(Row))>
`;
